addons/mail/static/src/core/common/composer.js

```diff
@@ -120,6 +123,10 @@ export class Composer extends Component {
         this.suggestion = this.store.self.type === "partner" ? useSuggestion() : undefined;
         this.markEventHandled = markEventHandled;
         this.onDropFile = this.onDropFile.bind(this);
+        this.saveContentDebounced = useDebounced(this.saveContent, 5000, {
+            execBeforeUnmount: true,
+        });
```

Comment:
Why do we debounce this at all? Does it cost a lot to save in the localstorage everytime the user input something?

Reply:
I've not checked but the answer is usually yes local storage is a disaster. On some browsers it might be a fully synchronous operations.

Reply:
Alright, but 5 full seconds? Could be a bit over kill no?

Reply:
5 seconds of inactivity usually means stop typing so a good time to save. It's the same value as the "typing" indicator